sources/kernel/syscalls.c

```diff
@@ -320,14 +340,51 @@ BrResult sys_ipc(BrIpcArgs *args)
     return result;
 }
 
-BrResult sys_irq(MAYBE_UNUSED BrIrqArgs *args)
+BrResult sys_irq(BrIrqArgs *args)
 {
     if (!(task_self()->caps & BR_CAP_IRQ))
     {
         return BR_BAD_CAPABILITY;
     }
 
-    return BR_NOT_IMPLEMENTED;
+    Task *task = nullptr;
+
+    if (args->task == BR_TASK_SELF)
+    {
+        task_ref(task_self());
+        task = task_self();
+    }
+    else
+    {
+        task = (Task *)domain_lookup(task_self()->domain, args->task, BR_OBJECT_TASK);
+    }
+
+    if (!task)
+    {
+        return BR_BAD_HANDLE;
+    }
+
+    auto irq = (Irq *)domain_lookup(task->domain, args->handle, BR_OBJECT_IRQ);
+
+    if (!irq)
+    {
+        return BR_BAD_HANDLE;
+    }
+
+    if (args->flags & BR_IRQ_ACK)
+    {
+        return irq_handler_ack(irq);
+    }
+    else if (args->flags & BR_IRQ_BIND)
+    {
+        return irq_handler_bind(task->handle, args->flags, irq, &args->msg);
+    }
+    else if (args->flags & BR_IRQ_UNBIND)
+    {
+        return irq_handler_unbind(args->flags, irq);
+    }
```

Comment:
Maybe these should be separated syscalls like:

- br_bind
- br_unbind
- br_ack